src/server/routes/entity/entity.js

```diff
@@ -773,6 +708,133 @@ export function handleCreateOrEditEntity(
+		return allEntities;
+	}
+	const relationships = relationshipSet.related('relationships').toJSON();
+
+	const otherEntitiesToFetch = relationships
+		.reduce((accumulator, relationship) => {
+			if (relationship.sourceBbid === entityBBID) {
+				accumulator.push(relationship.targetBbid);
+			}
+			else if (relationship.targetBbid === entityBBID) {
+				accumulator.push(relationship.sourceBbid);
+			}
+			return accumulator;
+		}, []);
+
+	await Promise.all(otherEntitiesToFetch.map(
+		async (bbid) => {
+			const otherEntity = await getEntityByBBID(orm, transacting, bbid);
+
+			const otherEntityRelationshipSet = await entity.relationshipSet()
+				.fetch({require: false, transacting, withRelated: 'relationships'});
+			if (!otherEntityRelationshipSet) {
+				return;
+			}
+			const otherEntityRelationships = otherEntityRelationshipSet.related('relationships').toJSON();
+
+			// Remove relationships with entity being deleted
+			const cleanedUpRelationships = otherEntityRelationships
+				.filter(({sourceBbid, targetBbid}) =>
+					entityBBID !== sourceBbid &&
+					entityBBID !== targetBbid);
+
+			// If there's a difference, apply the new relationships array.
+			if (cleanedUpRelationships.length !== otherEntityRelationships.length) {
+				await orm.func.relationship.updateRelationshipSets(
+					orm, transacting, otherEntityRelationshipSet, cleanedUpRelationships
+				);
+				// List of BBID of all the affected entities
+				allEntities = _.unionBy(allEntities, [otherEntity], 'id');
+			}
+		}
+	));
+
+	return allEntities;
+}
+
+export function handleDelete(
+	orm: any, req: PassportRequest, res: $Response, HeaderModel: any,
+	RevisionModel: any
+) {
+	// Destructuring entity from localsas entityJSON
+	const {entity: entityJSON}: {entity: any} = res.locals;
+	// const {entity}: {entity: any} = res.locals;
+	const {Revision, bookshelf} = orm;
+	const editorJSON = req.session.passport.user;
+	const {body}: {body: any} = req;
+
+	const entityDeletePromise = bookshelf.transaction(async (transacting) => {
+		const entity = await getEntityByBBID(orm, transacting, entityJSON.bbid);
+
+		await utils.incrementEditorEditCountById(orm, editorJSON.id, transacting);
+
+		const newRevision = new Revision({authorId: editorJSON.id});
+
+		await newRevision.save(null, {transacting});
+
+		const notePromise = _createNote(orm, body.note, editorJSON.id, newRevision, transacting);
+
+		entity.set('dataId', null);
+
+		// Update related Entities
+		const updatedEntities = await deleteEntityRelationships(orm, transacting, entity);
+
+		const parentRevisionIDs = _.compact(updatedEntities.map((entityModel) => entityModel.get('revisionId')));
+
+		await setParentRevisions(transacting, newRevision, _.uniq(parentRevisionIDs));
+
+		/*
+		 * No trigger for deletions, so manually create the <Entity>Revision
+		 * and update the entity header
+		 */
+
+		const entitiesRevisions = await Promise.all(
+			_.map(updatedEntities, (entityModel) => new RevisionModel({
```

Comment:
This isn't going to work (and neither is the `new HeaderModel(…` below).
The reason for that is RevisionModel and HeaderModel refer to a specific entity type.
For example, if I'm deleting an Author, it will be AuthorRevision and AuthorHeader.

Now if the Author I'm deleting has a relationship to a Work —for example— you're trying to create a new AuthorRevision and AuthorHeader with the Work's bbid (instead of a WorkRevision and WorkHeader).

That is currently throwing an error for me, and it's not what you want to do.
Instead, you would need to get the right model for the each updated entity.
The type (Author, Work…) you can get with `entityModel.get('type')`, and then use the `utils.getEntityModelByType` utility to fetch the right model.


However,  I think we can get away with not using the entity revision and header at all, and instead use the `saveEntitiesAndFinishRevision` utility that is used when editing entitites (have a look in the same file at how it is used in `handleCreateOrEditEntity`).
It also takes care of setting the parent revision ids, incrementing the editor count and creating a revision note, so it seems like a good idea to use that same utility instead of repeating code (a good refactor for free :) )

You'll have to try that out and see how it works out.
I know it's quite a heavy part of the website's mechanism, but you're doing a great job so far !
Good luck with the continuation :)